src/Workspaces/Core/Portable/Classification/SyntaxClassification/SyntacticChangeRangeComputer.cs

```diff
@@ -0,0 +1,220 @@
+            if (newStack.Count > 0)
+                return oldRoot.FullSpan.Length;
+
+            // We consumed both stacks entirely.  That means the trees were identical (though the root was different). Return null to signify no change to the doc.
+            return null;
+        }
+
+        private static int? ComputeCommonRightWidth(
+            SyntaxNode oldRoot,
+            SyntaxNode newRoot,
+            Stack<SyntaxNodeOrToken> oldStack,
+            Stack<SyntaxNodeOrToken> newStack,
+            CancellationToken cancellationToken)
+        {
+            while (oldStack.Count > 0 && newStack.Count > 0)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                var currentOld = oldStack.Pop();
+                var currentNew = newStack.Pop();
+
+                // The width on the right we've moved past on both old/new should be the same.
+                Contract.ThrowIfFalse((oldRoot.FullSpan.End - currentOld.FullSpan.End) ==
+                                      (newRoot.FullSpan.End - currentNew.FullSpan.End));
+
+                // If the two nodes/tokens were the same just skip past them.  They're part of the common right width.
+                if (currentOld.IsIncrementallyIdenticalTo(currentNew))
+                    continue;
+
+                // if we reached a token for either of these, then we can't break things down any further, and we hit
+                // the furthest point they are common.
+                if (currentOld.IsToken || currentNew.IsToken)
+                    return oldRoot.FullSpan.End - currentOld.FullSpan.End;
+
+                // we've got two nodes, but they weren't the same.  For example, say we made an edit in a method in the
+                // class, the class node would be new, but there might be many member nodes following the edited node
+                // that were the same that we'd want to see and skip.  Crumble the node and deal with its right side.
+                //
+                // Do not reverse the children.  We want to process the rightmost child first and walk right to left.
+                foreach (var nodeOrToken in currentOld.AsNode()!.ChildNodesAndTokens())
+                    oldStack.Push(nodeOrToken);
+
+                foreach (var nodeOrToken in currentNew.AsNode()!.ChildNodesAndTokens())
+                    newStack.Push(nodeOrToken);
+            }
+
+            // If we consumed all of 'new', then the length of the new doc is what we have in common.
+            if (oldStack.Count > 0)
+                return newRoot.FullSpan.Length;
+
+            // If we consumed all of 'old', then the length of the old doc is what we have in common.
+            if (newStack.Count > 0)
+                return oldRoot.FullSpan.Length;
+
+            // We consumed both stacks entirely.  That means the trees were identical (though the root was different). Return null to signify no change to the doc.
+            return null;
```

Comment:
💭 Is there any advantage to splitting these, as opposed to just returning `oldRoot.FullSpan.Length`?